src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -112,29 +113,29 @@ public void launch(SlaveComputer computer, TaskListener listener) {
             Pod pod = template.build(slave);
             slave.assignPod(pod);
 
-            String podId = pod.getMetadata().getName();
+            String podName = pod.getMetadata().getName();
 
             String namespace = Arrays.asList( //
                     pod.getMetadata().getNamespace(),
                     template.getNamespace(), client.getNamespace()) //
                     .stream().filter(s -> StringUtils.isNotBlank(s)).findFirst().orElse(null);
             slave.setNamespace(namespace);
 
-            LOGGER.log(Level.FINE, "Creating Pod: {0}/{1}", new Object[] { namespace, podId });
+            LOGGER.log(Level.FINE, "Creating Pod: {0}/{1}", new Object[] { namespace, podName });
             pod = client.pods().inNamespace(namespace).create(pod);
-            LOGGER.log(INFO, "Created Pod: {0}/{1}", new Object[] { namespace, podId });
-            listener.getLogger().printf("Created Pod: %s/%s%n", namespace, podId);
+            LOGGER.log(INFO, "Created Pod: {0}/{1}", new Object[] { namespace, podName });
+            listener.getLogger().printf("Created Pod: %s/%s%n", namespace, podName);
 
             TaskListener runListener = template.getListener();
-            runListener.getLogger().printf("Created Pod: %s in namespace %s%n", podId, namespace);
-
+            runListener.getLogger().printf("Created Pod: %s/%s%n", namespace, podName);
 
-            String podName = pod.getMetadata().getName();
-            String namespace1 = pod.getMetadata().getNamespace();
             template.getWorkspaceVolume().createVolume(client, pod.getMetadata());
             watcher = new AllContainersRunningPodWatcher(client, pod, runListener);
-            try (Watch _w = client.pods().inNamespace(namespace1).withName(podName).watch(watcher)) {
-                watcher.await(template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
+            try (Watch w1 = client.pods().inNamespace(namespace).withName(podName).watch(watcher);
+                 Watch w2 = eventWatch(client, podName, namespace, runListener)) {
+                if (watcher != null) {
```

Comment:
How could it be null? You just assigned it via a constructor. Does SpotBugs not reject this?

Reply:
I guess Spotbugs thinks `eventWatch` could have a side-effect.

Reply:
AFAICT the `watcher` field is only written to in the one place above, so it should be definitely non-null here. You can always

```suggestion
                assert watcher != null;
```

to make it behave.

Reply:
🤷‍♂